File: app/src/main/java/com/einthefrog/historyquiz/ui/MainMenuFragment.java
package com.einthefrog.historyquiz.ui;

import android.app.Activity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;

import com.einthefrog.historyquiz.R;
import com.einthefrog.historyquiz.databinding.FragmentMainMenuBinding;

public class MainMenuFragment extends Fragment {

    @Override
    public View onCreateView(
            LayoutInflater inflater,
            ViewGroup container,
            Bundle savedInstanceState
    ) {
        FragmentMainMenuBinding binding = FragmentMainMenuBinding.inflate(getLayoutInflater());

        Activity activity = getActivity();
        if (activity != null) {
            setOnStartClickListener(activity, binding);
        }

        return binding.getRoot();
    }

    private void setOnStartClickListener(Activity activity, FragmentMainMenuBinding binding) {
        NavController navController = Navigation.findNavController(activity, R.id.nav_host_fragment);
        binding.buttonStart.setOnClickListener(view -> {
            navController.navigate(R.id.action_mainMenuFragment_to_mainGameFragment);
        });
    }
}

File: app/src/main/java/com/einthefrog/historyquiz/ui/menu/MainMenuFragment.kt
package com.einthefrog.historyquiz.ui.menu

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import com.einthefrog.historyquiz.R
import com.einthefrog.historyquiz.databinding.FragmentMainMenuBinding

class MainMenuFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val binding: FragmentMainMenuBinding = FragmentMainMenuBinding.inflate(layoutInflater)
        val activity: Activity? = activity
        activity?.let { setOnStartClickListener(it, binding) }
        return binding.root
    }

    private fun setOnStartClickListener(activity: Activity, binding: FragmentMainMenuBinding) {
        val navController = findNavController(activity, R.id.nav_host_fragment)
        binding.buttonStart.setOnClickListener { navController.navigate(R.id.action_mainMenuFragment_to_mainGameFragment) }
    }
}
